apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexBeanProducer.java

```diff
@@ -16,41 +16,44 @@
 
 @Singleton
 public class DexBeanProducer {
-
-    private PropertiesHolder propertiesHolder;
-    private static Web3j web3j;
-    private static TransactionReceiptProcessor transactionReceiptProcessor;
+    private final PropertiesHolder propertiesHolder;
+    private static volatile Web3j web3j;
```

Comment:
I guess the web3j field should be neither static nor volatile.

Reply:
I reckon that it should volatile but not static. It explains in detail in the link provided yesterday